src/bproxy.c

```diff
@@ -170,9 +171,10 @@ void proxy_read_cb(uv_stream_t *handle, ssize_t nread, const uv_buf_t *buf)
     {
       log_error("could not read from socket!");
     }
-    if (uv_is_closing((const uv_handle_t *)handle))
+    if (!uv_is_closing((const uv_handle_t *)handle))
     {
       uv_close((uv_handle_t *)handle, proxy_close_cb);
+      conn->proxy_handle = NULL;
```

Comment:
shouldn't you need to `free(conn->proxy_handle);` also, not just setting the value to `NULL`?

Reply:
It should be freed in close callback. I'll check that.

Reply:
yes, its freed in cb, I missed that.